### PR TITLE
Adds Python 3.7 support to CI

### PR DESCRIPTION
**Pull request by eemperor**

### CODE REVIEW

**Comment by eemperor** [comment]
Oops.  Guess I should do what comment says...

File: .travis.yml [hunk 1]
```diff
@@ -54,6 +54,16 @@ notifications:
     on_failure: always
 jobs:
   include:
+    - stage: test
+      python: 3.7
+      dist: xenial
+      env:
+        - TOXENV=check
```
Comment (lorengordon) [inline .travis.yml]: I think we can remove the `env` block. We don't want to override TOXENV. It should be set by the top-level `before_install`.

File: .travis.yml [hunk 2]
```diff
@@ -54,6 +54,16 @@ notifications:
...
+    - stage: test
```
Comment (lorengordon) [inline .travis.yml]: This job can be removed. We aren't worried at the moment about the docs. We're really just getting at pytest.

File: .travis.yml [hunk 3]
```diff
@@ -87,7 +97,7 @@ jobs:
           tags: true
     - stage: deploy
       if: branch in (env(RELEASE_BRANCH), develop) OR tag is present
-      python: 3.6
+      python: 3.7
```
Comment (lorengordon) [inline .travis.yml]: Let's not change this... I think that will impact the standalone builds.

File: appveyor.yml [hunk 4]
```diff
@@ -14,10 +14,10 @@ init:
 install:
   - git submodule update --init --recursive
 test_script:
-  # ----- python 3.6, x64
+  # ----- python 3.7, x64
```
Comment (lorengordon) [inline appveyor.yml]: This also should be a _new_ test, not replacing the 3.6 tests...

**Comment by eemperor** [comment]
@lorengordon Updates made per your comments.  Ready for re-review.

**Pull request merged**